keyboards/viktus/minne_topre/ec.c

```diff
@@ -0,0 +1,181 @@
+/* Copyright 2023 Viktus Design LLC
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include "quantum.h"
+#include "ec.h"
+#include "analog.h"
+//#include "debug.h"  // needed for debugging
+
+// sensing channel definitions
+#define A0 0
+#define A1 1
+#define A2 2
+#define A3 3
+#define A4 4
+#define A5 5
+#define A6 6
+#define A7 7
+
+// analog connection settings
+#define DISCHARGE_PIN D5
+#define ANALOG_PORT D4
+
+#ifndef MUX_SEL_PIN
+#define MUX_SEL_PINS \
+        { D1, D2, D3 }
+#endif
+
+// pin connections
+const uint8_t row_channels[] = MATRIX_ROW_PINS;
+const uint8_t col_pins[]     = MATRIX_COL_PINS;
+const uint8_t mux_sel_pins[] = MUX_SEL_PINS;
+
+_Static_assert(sizeof(mux_sel_pins) == 3, "invalid MUX_SEL_PINS");
+
+static ec_config_t config;
+static uint16_t    ec_sw_value[MATRIX_COLS][MATRIX_ROWS];
+
+static inline void discharge_capacitor(void) { setPinOutput(DISCHARGE_PIN); }
+static inline void charge_capacitor(uint8_t col) {
+    setPinInput(DISCHARGE_PIN);
+    writePinHigh(col_pins[col]);
+}
+
+static inline void clear_all_col_pins(void) {
+    for (int col = 0; col < sizeof(col_pins); col++) {
+        writePinLow(col_pins[col]);
+    }
+}
+
+void init_mux_sel(void) {
+    for (int idx = 0; idx < sizeof(mux_sel_pins); idx++) {
+        setPinOutput(mux_sel_pins[idx]);
+    }
+}
+
+void select_mux(uint8_t row) {
+    uint8_t ch = row_channels[row];
+    writePin(mux_sel_pins[0], ch & 1);
+    writePin(mux_sel_pins[1], ch & 2);
+    writePin(mux_sel_pins[2], ch & 4);
+}
+
+void init_col(void) {
+    for (int idx = 0; idx < sizeof(col_pins); idx++) {
+        setPinOutput(col_pins[idx]);
+        writePinLow(col_pins[idx]);
+    }
+}
+
+int ec_init(ec_config_t const* const ec_config) {
```

Comment:
This only returns 0.  If that's the case, does this even need to be typed as int?  wouldn't `void` be much more accurate/useful. 

```suggestion
void ec_init(ec_config_t const* const ec_config) {
```